NeoMathEngine/src/MathEngineDnnDropout.h

```diff
@@ -15,24 +15,30 @@ limitations under the License.
 
 #pragma once
 
-#include <NeoMathEngine/NeoMathEngine.h>
+#include <NeoMathEngine/MathEngineDropout.h>
 #include <NeoMathEngine/CrtAllocatedObject.h>
 
 namespace NeoML {
 
-// Dropout descriptor
-struct CMathEngineDropoutDesc : public CDropoutDesc {
-	explicit CMathEngineDropoutDesc( IMathEngine& mathEngine, float rate, bool isSpatial, bool isBatchwise,
-		const CBlobDesc& input, const CBlobDesc& output, int seed );
-
-	CBlobDesc Input; // input blob descriptor
-	CBlobDesc Output; // output blob descriptor
-	const float ForwardRate; // the probability that an element is not dropped out
-	const bool IsSpatial; // indicates if whole channels are dropped out
-	const bool IsBatchwise; // indicates if an element is dropped out of all objects in one batch at the same time
-	// A blob that stores the dropout information for each element on the last run
-	// Only used when learning
-	CFloatHandleVar Mask;
+static inline int getMaskSize(bool isSpatial, bool isBatchwise, const CBlobDesc& input)
```

Comment:
Super-minor comment: `getMaskSize` could be static method of `CMaskDropoutDesc` to show its need for that only. 
Or stay here, on your teste